### DISCUSSION TITLE
Help with Hello World Agents.Net

### DISCUSSION THREAD

**Discussion by dawningtruth** [agents-net/agents.net#114] [Q&A]
Hi,

I have tried to get the Hello World working for Agents.Net

It is not working. I think there is something missing. Would it be possible for you to take a quick look and advice what I have left out or done wrong.

https://www.dropbox.com/sh/n99j1sundgnijsh/AAAKT4c0PtreLV3Qg3ZIuHX_a?dl=0

Thx

**Comment by twilker** [comment]
Hi there,

Thanks for trying out Agents.Net. I have looked at your code and it look really good. I think you forgot to include the message classes for `HelloConsoleMessage, WolrdConsoleMessage and ConsoleMessageCreated` into the solution you shared. Once I added those to the project there was only one thing still missing and that is in the Program.Main method you need to add the following line
`finishedEvent.WaitOne();`
This will wait until the agent framework is finished. Otherwise the program is terminated before the console message is created as everything in Agents.Net is executed in its own thread.

If you have any other question feel free to ask me. I am using it already in all of my projects and know a pitfall or two ;-)

Kind regards.
